src/main/java/com/softserveinc/dokazovi/aop/PostLogger.java

```diff
@@ -96,16 +88,11 @@ public Boolean updatePost(ProceedingJoinPoint proceedingJoinPoint) throws Throwa
                 case "PUBLISHED":
                     changes = "Опубліковано";
                     break;
-                default: changes = "N/A";
+                default:
+                    changes = "N/A";
             }
         }
-        LogEntity log = LogEntity.builder()
-                .title(postSaveFromUserDTO.getTitle())
-                .changes(changes)
-                .nameOfChanger(userEntity.getLastName() + " " + userEntity.getFirstName())
-                .build();
-
-        logRepository.save(log);
+        makeEntryInLogs(postSaveFromUserDTO.getTitle(), userPrincipal, changes);
         return joinPoint;
     }
 
```

Comment:
You can also avoid loops like this
```
 for (Object obj : arguments) {
            if (obj instanceof UserPrincipal) {
                userPrincipal = (UserPrincipal) obj;
            }
            if (obj instanceof Integer) {
                postId = (Integer) obj;
            }
            if (obj instanceof Boolean) {
                flag = (boolean) obj;
            }
        }
```

by assigning values directly from arguments array by filtering them by type like so
```
UserPrincipal userPrincipal = (UserPrincipal) Arrays.stream(arguments)
                .filter(obj -> obj instanceof UserPrincipal)
                .findFirst().orElse(null);
```